code/controllers/subsystems/job.dm

```diff
@@ -730,15 +730,7 @@
 
 	if(PDA_Manifest.len)
 		PDA_Manifest.Cut()
-	for(var/datum/data/record/R in data_core.medical)
-		if ((R.fields["name"] == H.real_name))
-			qdel(R)
-	for(var/datum/data/record/T in data_core.security)
-		if ((T.fields["name"] == H.real_name))
-			qdel(T)
-	for(var/datum/data/record/G in data_core.general)
-		if ((G.fields["name"] == H.real_name))
-			qdel(G)
+		qdel(SSrecords.find_record("name", H.real_name))
```

Comment:
This leaves a null entry in the list. My suggestion is to implement a `SSrecords.delete_record(field, subject)`, which would also remove the entry from the list before qdel-ing.